stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/WebSocketServerConnectorListener.java

```diff
@@ -127,12 +126,16 @@ public void notifySuccess() {
                         Resource onOpenResource = wsService.getResourceByName(WebSocketConstants.RESOURCE_NAME_ON_OPEN);
                         WebSocketOpenConnectionInfo connectionInfo =
                                 connectionManager.getConnectionInfo(webSocketInitMessage.getSessionID());
-                        WebSocketConnection webSocketConnection = connectionInfo.getWebSocketConnection();
-                        if (onOpenResource != null) {
-                            BStruct webSocketEndpoint = connectionInfo.getWebSocketEndpoint();
-                            WebSocketUtil.executeOnOpenResource(onOpenResource, webSocketEndpoint, webSocketConnection);
-                        } else {
-                            connectionInfo.getWebSocketConnection().readNextFrame();
+                        WebSocketConnection webSocketConnection;
+                        if (connectionInfo != null &&
```

Comment:
IMO we should not check this since if the flow is correct connectionInfo should be there and if we add if condition here then this will skip the next steps. If we really need this then we should throw an error rather than going forward. WDYT?